compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1013,8 +1013,15 @@ impl EncodeContext<'a, 'tcx> {
         self.encode_inferred_outlives(def_id);
 
         // This should be kept in sync with `PrefetchVisitor.visit_trait_item`.
```

Comment:
Is this comment still relevant?

Because I cannot tell from looking at the body of the referenced fn: 

https://github.com/rust-lang/rust/blob/8150ad8abc1a4cf378a71e14356c582c09f93d28/compiler/rustc_metadata/src/rmeta/encoder.rs#L1989-L1991

whether these two things are "in sync" or not.

Reply:
You are looking at the wrong commit. The right one is https://github.com/rust-lang/rust/blob/76ab8ba274f631cf5ae5748c95f12aaf0e5ce695/compiler/rustc_metadata/src/rmeta/encoder.rs#L2025-L2037

Reply:
I am not sure whether we need to also have a `prefetch_promoted_mir`, we didn't have it before, so that should probably be a separate PR

Reply:
wow, that's some egg on my face. Its very polite of you to not point out that the *right* commit is in fact **this** commit that I'm commenting on.

Reply:
well... the commits in this PR are not very organized... There is some back and forth, I should have cleaned those up before asking for a review. Basically I noticed that I need to sync the `visit_trait_item` function several commits after the commit that you were reviewing, I should merge it back into the original commit so the sync happens at the same time